clean_project/project/index.py

```diff
@@ -39,12 +44,14 @@ def index(config_path):
     if data is not None:
         index_words = defaultdict(lambda: defaultdict(list))
         for file_dirty in data['files_path']:
-            with codecs.open(file_dirty.get('file'), 'r', 'utf-8') as source_file:
+            opener = urllib.urlopen(file_dirty.get('file'))
+            with contextlib.closing(opener) as source_file:
+                read_source_file = source_file.readlines()
```

Comment:
don't use `readlines()`, as it creates a list with all the lines in the file. if the file is very big, you could run out of memory